command/state_show.go

```diff
@@ -45,22 +45,27 @@ func (c *StateShowCommand) Run(args []string) int {
 		return 1
 	}
 
+	if len(results) == 0 {
+		return 0
+	}
+
 	instance, err := c.filterInstance(results)
 	if err != nil {
 		c.Ui.Error(err.Error())
 		return 1
 	}
+
 	is := instance.Value.(*terraform.InstanceState)
 
 	// Sort the keys
-	keys := make([]string, 0, len(is.Attributes))
```

Comment:
It seems like you _hate_ performance!?!? 😆 

The allocs, think of the allocs!!! 😛